### ISSUE TITLE
Add API key names/description field

### ISSUE THREAD

**Issue by KetchupBomb** [shlinkio/shlink#1044]
#### Summary

```
# This is example data -- don't share your actual API keys!
$ shlink api-key:list
+--------------------------------------+------------+-----------------+-------+
| Key                                  | Is enabled | Expiration date | Roles |
+--------------------------------------+------------+-----------------+-------+
| 7797daee-5ddb-4e1e-b7ba-6dae81a35c77 | +++        | -               | Admin |
| 0140df02-45f3-4df7-ab71-c1ba10bea1cb | +++        | -               | Admin |
| f8604111-ce6f-48b4-bc32-ed144207559f | +++        | -               | Admin |
+--------------------------------------+------------+-----------------+-------+
```

It's easy to see what keys have been generated, but there's no DB-based "tracking" mechanism to know in which context each key is used.

For instance, if I install the Shlink Chrome extension on my personal and work computers, there's no easy way to tell them apart from the server-side. I'd have to check the configuration of each Chrome extension.

I suggest adding the ability to add a(n optional, for backward compatibility & historical low-friction semantics) "name" or "description" in order to allow administrators to control pseudo-authentication:

```
# This is example data -- don't share your actual API keys!
$ shlink api-key:generate --name 'Windows PC'
$ shlink api-key:list
+--------------------------------------+------------+-----------------+-------+------------+
| Key                                  | Is enabled | Expiration date | Roles | Name       |
+--------------------------------------+------------+-----------------+-------+------------+
| 7797daee-5ddb-4e1e-b7ba-6dae81a35c77 | +++        | -               | Admin |            |
| 0140df02-45f3-4df7-ab71-c1ba10bea1cb | +++        | -               | Admin |            |
| f8604111-ce6f-48b4-bc32-ed144207559f | +++        | -               | Admin |            |
| e658a8aa-82a6-4a5d-b117-410343ce8bc4 | +++        | -               | Admin | Windows PC |
+--------------------------------------+------------+-----------------+-------+------------+
```

Even if this information is only stored in the DB for now, its usefulness is immediate for administrators.

**Comment by KetchupBomb** [comment]
I'm working on a diff right now. Running into unit testing issues, but my code is functioning empirically.

There are some matters of opinion that I'll call out when I put up the PR for review.

**Comment by acelaya** [comment]
Thanks!

Just one thing that I can see from your example. I would maybe put the name column in the first or second place of the table.

Regarding the tests, if you can't come around them, feel free to create the PR anyway, and we can continue from there.

**Comment by KetchupBomb** [comment]
> @acelaya: Just one thing that I can see from your example. I would maybe put the name column in the first or second place of the table.

You already spotted one of the "matters of opinion" I wanted to bring up. We agree on this point, but there's more context to discuss, too, like DB column placement.

> @acelaya: Regarding the tests, if you can't come around them, feel free to create the PR anyway, and we can continue from there.

Cool, I was hoping you'd be ok with this. I'll point out the problems I'm having within the incoming PR.

**Comment by acelaya** [comment]
> like DB column placement

At the end of the table is ok.

**Comment by acelaya** [comment]
Closed by #1045
